### PR TITLE
Arjuns/softmax loss documentation update

### LINKED ISSUE #2302: Improve documentation of _SoftmaxLoss
more details at: https://github.com/allenai/allennlp/pull/2264#discussion_r245162745

### PR DESCRIPTION
**Pull request by ArjunSubramonian**
Fixes #2302 .

Changes proposed in this pull request:

- Documentation update regarding padding made to SoftmaxLoss.
- **I'm unsure if this documentation update makes sense**

### CODE REVIEW

**Comment by dirkgr** [comment]
I don't really understand this. What's "adding a padding ID"?

**Comment by ArjunSubramonian** [comment]
@dirkgr Here is an excerpt from the issue that explains it well:

"I suppose it could be reasonable to assume that a class such as this might add a word into the vocabulary corresponding to a padding ID (i.e. new vocabulary size would be num_words+1), and that the target tensor argument could include such an id."

**Comment by dirkgr** [comment]
Ah, got it. Can you add "into the vocabulary" to the new docstring?

File: allennlp/modules/softmax_loss.py [hunk 1]
```diff
@@ -22,6 +23,20 @@ def __init__(self, num_words: int, embedding_dim: int) -> None:
...
+            A tensor of shape `(sequence_length, embedding_dim)`
+        targets : `torch.Tensor`
+            A tensor of shape `(batch_size, )`
+
+        # Returns
+
+        loss : `torch.FloatTensor`
+            A scalar loss to be optimized.
+        """
+
```
Comment (epwalsh) [inline allennlp/modules/softmax_loss.py:39]: Just removing extra empty lines

```suggestion
        """
        # Parameters

        embeddings : `torch.Tensor`
            A tensor of shape `(sequence_length, embedding_dim)`
        targets : `torch.Tensor`
            A tensor of shape `(batch_size, )`

        # Returns

        loss : `torch.FloatTensor`
            A scalar loss to be optimized.
        """
```

**Review by epwalsh — DISMISSED** [review]
LGTM!

**Review by epwalsh — APPROVED** [review]
✅

**Pull request merged**